### PR TITLE
Use id to identify list

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

**Comment by JacquesBorsenberger** [comment]
depends on 
https://github.com/gridsuite/gridstudy-app/pull/363
https://github.com/gridsuite/gridactions-app/pull/47
https://github.com/gridsuite/filter-server/pull/5

**Review by niconoir — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java [hunk 1]
```diff
@@ -58,82 +58,99 @@ public ContingencyListController(ContingencyListService service) {
...
-    public ResponseEntity<Void> deleteContingencyList(@PathVariable("name") String name) {
-        service.deleteContingencyList(name);
+    public ResponseEntity<Void> deleteContingencyList(@PathVariable("id") UUID id) {
+        service.deleteContingencyList(id);
         return ResponseEntity.ok().build();
     }
 
-    @PostMapping(value = "contingency-lists/{name}/rename")
-    @ApiOperation(value = "Rename contingency list by name")
+    @PostMapping(value = "contingency-lists/{id}/rename")
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/ContingencyListController.java]: removed finction rename is an update (put)

File: src/main/java/org/gridsuite/actions/server/repositories/FiltersContingencyListRepository.java [hunk 2]
```diff
@@ -10,25 +10,25 @@
...
  * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
  */
 @Repository
-public interface FiltersContingencyListRepository extends JpaRepository<FiltersContingencyListEntity, String> {
+public interface FiltersContingencyListRepository extends JpaRepository<FiltersContingencyListEntity, UUID> {
 
     @Query("SELECT DISTINCT entity FROM FiltersContingencyListEntity entity LEFT JOIN FETCH entity.countries")
     List<FiltersContingencyListEntity> findAllWithCountries();
 
     Optional<FiltersContingencyListEntity> findByName(String name);
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/repositories/FiltersContingencyListRepository.java]: removed

File: src/main/java/org/gridsuite/actions/server/repositories/FiltersContingencyListRepository.java [hunk 3]
```diff
@@ -10,25 +10,25 @@
...
 
-    @Transactional
-    void deleteByName(String name);
-
     boolean existsByName(String name);
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/repositories/FiltersContingencyListRepository.java]: removed

File: src/main/java/org/gridsuite/actions/server/repositories/FiltersContingencyListRepository.java [hunk 4]
```diff
@@ -10,25 +10,25 @@
...
+
+    /* do not use deleteById which throw when id does not exists */
+    Integer deleteFiltersContingencyListEntityById(UUID id);
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/repositories/FiltersContingencyListRepository.java:28]: It was a mistake, now I use is as intended.

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java [hunk 5]
```diff
@@ -99,318 +104,400 @@ public void setUp() {
...
-                .content(content)
-                .contentType(APPLICATION_JSON))
-                .andExpect(status().isOk());
+        String res = mvc.perform(post("/" + VERSION + "/filters-contingency-lists/")
+            .content(content)
+            .contentType(APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andReturn().getResponse().getContentAsString();
 
+        //JSONAssert.assertEquals(content, res, JSONCompareMode.LENIENT);
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java]: was use for debuging forgot to remove it; now there is functions to compare filter

**Review by jonenst — APPROVED** [review]

**Pull request merged**